x-pack/test/functional/services/ml/deploy_models_flyout.ts

```diff
@@ -0,0 +1,296 @@
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+      // FIELD MAP
+      const editFieldMapButton = await testSubjects.find(
+        'mlTrainedModelsInferencePipelineFieldMapEditButton'
+      );
+      await editFieldMapButton.click();
+      await this.setTrainedModelsInferenceFlyoutCustomEditorValues(
+        'mlTrainedModelsInferencePipelineFieldMapEdit',
+        JSON.stringify(values.editedFieldMap)
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+      // OPEN ADVANCED SETTINGS
+      await this.trainedModelsInferenceOpenAdditionalSettings();
+      await this.setTrainedModelsInferenceFlyoutCustomValue(
+        'mlTrainedModelsInferenceAdvancedSettingsConditionTextArea',
+        values.condition
+      );
+      await this.setTrainedModelsInferenceFlyoutCustomValue(
+        'mlTrainedModelsInferenceAdvancedSettingsTagInput',
+        values.tag
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+
+      await this.assertTrainedModelsInferenceFlyoutPipelineConfigValues(
+        values.editedInferenceConfig,
+        values.editedFieldMap
+      );
+      await this.assertTrainedModelsInferenceFlyoutAdditionalSettings(values.condition, values.tag);
+    }
+
+    public async completeTrainedModelsInferenceFlyoutDetails(
+      expectedValues: {
+        name: string;
+        description: string;
+        targetField: string;
+      },
+      editDefaults: boolean = false
+    ) {
+      if (editDefaults) {
+        await this.setTrainedModelsInferenceFlyoutCustomDetails(expectedValues);
+      }
+      const name = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineNameInput',
+        'value'
+      );
+      expect(name).to.eql(expectedValues.name);
+      const description = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineDescriptionInput',
+        'value'
+      );
+      expect(description).to.eql(expectedValues.description);
+      const targetField = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineTargetFieldInput',
+        'value'
+      );
+      expect(targetField).to.eql(expectedValues.targetField);
+      await this.deployModelsContinue('mlTrainedModelsInferencePipelineProcessorConfigStep');
+    }
+
+    public async trainedModelsInferenceOpenAdditionalSettings() {
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
+      await testSubjects.click('mlTrainedModelsInferenceAdvancedSettingsAccordionButton');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsConditionTextArea');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsTagInput');
+    }
+
+    public async completeTrainedModelsInferenceFlyoutPipelineConfig(
+      expectedValues: {
+        inferenceConfig: any;
+        editedInferenceConfig?: any;
+        fieldMap: any;
+        editedFieldMap?: any;
```

Comment:
could you please provide types here?

Reply:
Added in 001a8301d6190c0a994c65cff3a18105aeda53b4